.buildkite/pipeline.yml

```diff
@@ -44,16 +44,24 @@ steps:
       build.env("K8S_NIGHTLY") == "1"
     steps:
       - key: charts-tests
-        label: Charts tests
+        label: Charts tests {{matrix.testpackagename}}
         timeout_in_minutes: 20
         notify:
           - github_commit_status:
               context: charts-tests
         commands:
           - |
-            TAG_NAME=$(ci/scripts/tag-check.sh) ./ci/scripts/run-in-nix-docker.sh task ci:charts:tests
+            TAG_NAME=$(ci/scripts/tag-check.sh) ./ci/scripts/run-in-nix-docker.sh task ci:unit-tests:{{matrix.testpackagename}}
         agents:
           queue: v6-amd64-builders
+        matrix:
+          setup:
+            testpackagename:
```

Comment:
See previous note about using a more general runner: `task ci:test PKG=PKG -- --extra-flags-here`

It's be _super_ nice if we didn't have to keep this up to date ourselves. What would you think about adding a tiny `yq` script to `task generate` that would set this field to the output of `go list -f '{{.Dir}}/...' -m` (See task lint for an explanation). That way we'd have a single pipeline for all go tests and an easy way of maintaining it.

Reply:
The problem for me is not maintaining the CI scripts, but the failure of one test put's entire PR delayed by one hour as we have very flaky CI. I would like to have fine grain CI pipeline that would work to my advantage, so each stage potentially can run in parallel and when failure occurs, we need to re-try only small step.

Reply:
With your suggestion I think the solution should be exactly what you described.